common/src/main/java/com/wynntils/handlers/inventory/InventoryHandler.java

```diff
@@ -0,0 +1,186 @@
+                if (!heldStack.isEmpty()) {
+                    running = new RunningInteraction(menu, (event.getMouseButton() & 0x4) != 0, heldStack.copy());
+                }
+            }
+            case PICKUP_ALL -> {
+                if (!heldStack.isEmpty()) {
+                    expect(new PendingInteraction.PickUpAll(menu, heldStack.copy()));
+                }
+            }
+                // SWAP and CLONE are not supported as inventory operations on Wynncraft
+        }
+    }
+
+    @SubscribeEvent
+    public void onSetContents(ContainerSetContentEvent.Pre event) {
+        long now = System.currentTimeMillis();
+        int containerId = event.getContainerId();
+        List<ItemStack> newInv = event.getItems();
+        pending.removeIf(entry -> now >= entry.expiryTime || !entry.interaction.onContentUpdate(containerId, newInv));
+    }
+
+    @SubscribeEvent
+    public void onSetSlot(ContainerSetSlotEvent.Pre event) {
+        long now = System.currentTimeMillis();
+        int containerId = event.getContainerId();
+        int slotNum = event.getSlot();
+        ItemStack newStack = event.getItemStack();
+        pending.removeIf(
+                entry -> now >= entry.expiryTime || !entry.interaction.onSlotUpdate(containerId, slotNum, newStack));
+    }
+
+    @SubscribeEvent
+    public void onWorldStateChange(WorldStateEvent event) {
+        running = null;
+        pending.clear();
+    }
+
+    public void forceSync(AbstractContainerMenu menu) {
+        McUtils.sendPacket(new ServerboundContainerClickPacket(
+                menu.containerId,
+                menu.getStateId(),
+                Math.min(1, menu.slots.size() - 1),
+                0,
+                ClickType.SWAP,
+                new ItemStack(Items.BARRIER, 31),
+                Int2ObjectMaps.emptyMap()));
+    }
+
+    private void forceSyncLater(AbstractContainerMenu menu) {
+        Managers.TickScheduler.scheduleNextTick(() -> forceSync(menu));
+    }
+
+    private void expect(PendingInteraction interaction) {
+        pending.add(new PendingInteractionEntry(interaction));
+    }
+
+    private final class RunningInteraction {
+        private final AbstractContainerMenu menu;
+        private final boolean single;
+        private final ItemStack heldStack;
+        private final IntList slots = new IntArrayList();
+
+        private RunningInteraction(AbstractContainerMenu menu, boolean single, ItemStack heldStack) {
+            this.menu = menu;
+            this.single = single;
+            this.heldStack = heldStack;
+        }
+
+        private boolean handle(ContainerClickEvent event) {
+            if (event.getClickType() != ClickType.QUICK_CRAFT
+                    || event.getContainerMenu().containerId != menu.containerId) {
+                running = null;
+                return false;
+            }
+
+            int mask = event.getMouseButton();
+            if (((mask & 0x4) == 0) == single) {
```

Comment:
Please use constants instead of magic numbers in code. (Also applies to line 161 below.)